### ISSUE TITLE
compiler error with protobuf 3.5.1for cpp 

### ISSUE THREAD

**Issue by Hakunametata** [protocolbuffers/protobuf#4276]
there is error description 

Error C4146： unary minus operator applied to unsigned type, result still unsigned   wire_format_lite.h	863	

Error C4146： unary minus operator applied to unsigned type, result still unsigned   wire_format_lite.h	874

**Comment by s-bear** [comment]
I was just investigating this as well. Fresh install using `vcpkg install protobuf[zlib] protobuf[zlib]:x64-windows` as per [https://github.com/google/protobuf/blob/master/src/README.md#c-installation---windows](https://github.com/google/protobuf/blob/master/src/README.md#c-installation---windows).

It seems like this is known, to some extent: [https://github.com/google/protobuf/tree/master/cmake#notes-on-compiler-warnings](https://github.com/google/protobuf/tree/master/cmake#notes-on-compiler-warnings) but this isn't just a warning, it's an error with the default MSVC settings.

You have to disable SDL Checks in the project properties to have it treated as just a warning again. [https://msdn.microsoft.com/en-us/library/jj161081.aspx?f=255&MSPPError=-2147217396](https://msdn.microsoft.com/en-us/library/jj161081.aspx?f=255&MSPPError=-2147217396).

This should be in the documentation, or perhaps a `#pragma warning (suppress :  4146 )` could be added to suppress this warning on the relevant lines?

**Comment by ZeldaZach** [comment]
Been getting hit with tons of these C4146 warnings. Is this a regression or has it always been like this? (Can see my Appveyor build [here](https://ci.appveyor.com/project/ZeldaZach/cockatrice/build/2.4.1-branch-fix_appveyor_stuff-build-404/job/pm9fhjiq91qdi91q))

**Comment by timou** [comment]
If you have a cmake build, and you want to suppress this, you can do the following. This is kind of annoying, because it propagates the suppression on to anything that links libprotobuf (which could mask real issues), but can work around the warnings until it's fixed.

```cmake
if( MSVC )
    target_compile_options( libprotobuf PUBLIC "/wd4146" )
endif() # MSVC
```

**Comment by TeBoring** [comment]
Could you send us a simple PR?

**Comment by TeBoring** [comment]
Seems this has been fixed by #4000
